R/tribble.R

```diff
@@ -27,19 +27,75 @@
 #'   "b", 4:6
 #' )
 tribble <- function(...) {
+  data <- extract_frame_data_from_dots(...)
 
+  if (length(data$frame_rest) == 0) {
+    out <- rep(list(logical()), length(data$frame_names))
+    names(out) <- data$frame_names
+    return(as_tibble(out))
+  }
+
+  frame_mat <- matrix(data$frame_rest, ncol = length(data$frame_names), byrow = TRUE)
+
+  frame_col <- lapply(seq_len(ncol(frame_mat)), function(i) {
```

Comment:
It's based on the principles of "clean code" I learned from Robert C. Martin's book (not comprehensive):

- [x] Each function should do one thing (and do it well)
- [x] The purpose of the function should be obvious from its name
- [ ] Prefer short pure functions with few arguments
- [ ] All code in a function should operate on the same level of abstraction, so that the logic of each function is easy to understand

In addition, by extracting a function you could write a unit test specifically for that function, but that's currently not top priority.